### PR TITLE
Course search

### PR DESCRIPTION
**Pull request by mbareta**

### CODE REVIEW

File: lms/templates/courseware/search.html [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+<%block name="bodyclass">view-in-course view-course-info ${course.css_class or ''}</%block>
+
+<script type="text/javascript">
+  function handleSearch() {
+    var searchQuery = $('#searchQuery').val();
+    var recentlyViewedFilter = $('#recentlyViewedFilter').hasClass('active');
+    var bookmarkedFilter = $('#bookmarkedFilter').hasClass('active');
+
+    var search = 'query=' + searchQuery;
+    search += '&recentlyViewed=' + recentlyViewedFilter;
```
Comment (MiroDojkic) [inline lms/templates/courseware/search.html]: You can break this into lines without new assignment with + operator

File: lms/djangoapps/courseware/views/views.py [hunk 2]
```diff
@@ -392,6 +392,59 @@ def bookmarks(request, course_id):
 
     return render_to_response('courseware/bookmarks.html', context)
 
+@ensure_csrf_cookie
+@ensure_valid_course_key
+def search(request, course_id):
+    from search.api import perform_search
```
Comment (vukoviciv) [inline lms/djangoapps/courseware/views/views.py]: why import here?

File: lms/templates/courseware/search.html [hunk 3]
```diff
@@ -0,0 +1,115 @@
+<%page expression_filter="h"/>
```
Comment (vukoviciv) [inline lms/templates/courseware/search.html]: can you extract this file in fastTrac theme?

File: lms/templates/courseware/search.html [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    var recentlyViewedFilter = $('#recentlyViewedFilter').hasClass('active');
+    var bookmarkedFilter = $('#bookmarkedFilter').hasClass('active');
+
+    var search = 'query=' + searchQuery;
+    search += '&recentlyViewed=' + recentlyViewedFilter;
+    search += '&bookmarked=' + bookmarkedFilter;
+
+    window.location.search = search;
+  }
+</script>
```
Comment (vukoviciv) [inline lms/templates/courseware/search.html]: Is there a reason why script tag is opening and closing for each function? or is this legacy.... :smile:

File: lms/templates/courseware/search.html (hunk 3 above)
Comment (mbareta) [reply lms/templates/courseware/search.html]: This is a new file in edx. I can copy it over to FT theme but I suppose edx core should have it.

**Pull request merged**